Makefile

```diff
@@ -134,3 +147,63 @@ dep-status:
 .PHONY: dep-prune
 dep-prune:
 	dep prune -v
+
+## --------------------------------------
+## Docker
+## --------------------------------------
+
+.PHONY: docker-build
+docker-build: ## Build the docker image for controller-manager
+	docker build --pull --build-arg ARCH=$(ARCH) -f build/Dockerfile -t $(CONTROLLER_IMG)-$(ARCH):$(TAG) .
+	MANIFEST_IMG=$(CONTROLLER_IMG)-$(ARCH) MANIFEST_TAG=$(TAG) $(MAKE) set-manifest-image
+	$(MAKE) set-manifest-pull-policy
+
+.PHONY: docker-push
+docker-push: ## Push the docker image
+	docker push $(CONTROLLER_IMG)-$(ARCH):$(TAG)
```

Comment:
for this to work with quay.io, we would need to create a new repository on quay.io since the image would be renamed from baremetal-operator to baremetal-operator-amd64. If that is the only architecture supported for now, I don't really see the point of renaming though. That could be done when other architectures are introduced.